tests/v2/validation/provisioning/rke2/templates_test.go

```diff
@@ -0,0 +1,164 @@
+package rke2
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	v1 "github.com/rancher/rancher/pkg/apis/catalog.cattle.io/v1"
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/charts"
+	"github.com/rancher/shepherd/extensions/cloudcredentials"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/defaults"
+	"github.com/rancher/shepherd/extensions/projects"
+	"github.com/rancher/shepherd/extensions/provisioning"
+	"github.com/rancher/shepherd/extensions/users"
+	password "github.com/rancher/shepherd/extensions/users/passwordgenerator"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/rancher/shepherd/pkg/wait"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+const (
+	localCluster   = "local"
+	fleetNamespace = "fleet-default"
+)
+
+type ClusterTemplateTest struct {
+	Repo             *v1.ClusterRepo `json:"repo,omitempty" yaml:"repo,omitempty"`
+	TemplateName     string          `json:"templateName,omitempty" yaml:"templateName,omitempty"`
+	TemplateProvider string          `json:"templateProvider,omitempty" yaml:"templateProvider,omitempty"`
+}
+
+type ClusterTemplateTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	standardUserClient *rancher.Client
+	session            *session.Session
+	templateConfig     *ClusterTemplateTest
+	cloudCredentials   *cloudcredentials.CloudCredential
+}
+
+func (r *ClusterTemplateTestSuite) TearDownSuite() {
+	r.session.Cleanup()
+}
+
+func (r *ClusterTemplateTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	r.session = testSession
+	r.templateConfig = new(ClusterTemplateTest)
+	config.LoadConfig("templateTest", r.templateConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+	r.client = client
+
+	enabled := true
+	var testuser = namegenerator.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(r.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
```

Comment:
You should be able to omit the standard user altogether since you aren't using it in the test at all.